File: DSA/String/Code/ConcatenatingTwoStrings.py
# Concatenating Two Strings in Python
# Join two strings together

str1 = "Hello"
str2 = "World"

# Method 1: Using + operator
print("Method 1: Using + operator")
result = str1 + " " + str2
print("Result:", result)

# Method 2: Using f-string
print("\nMethod 2: Using f-string")
result = f"{str1} {str2}"
print("Result:", result)

# Method 3: Using .format()
print("\nMethod 3: Using .format()")
result = "{} {}".format(str1, str2)
print("Result:", result)

# Method 4: Using % formatting
print("\nMethod 4: Using % formatting")
result = "%s %s" % (str1, str2)
print("Result:", result)

# Method 5: Using .join()
print("\nMethod 5: Using .join()")
result = " ".join([str1, str2])
print("Result:", result)

# Method 6: Multiple strings
print("\nMethod 6: Multiple strings")
str3 = "from"
str4 = "Python"
result = str1 + " " + str2 + " " + str3 + " " + str4
print("Result:", result)

# Method 7: Using .join() with multiple strings
print("\nMethod 7: .join() with multiple")
result = " ".join([str1, str2, str3, str4])
print("Result:", result)

# Method 8: Using list comprehension and join
print("\nMethod 8: Dynamic concatenation")
strings = ["Hello", "World", "from", "Python"]
result = " ".join(strings)
print("Result:", result)

# Method 9: Concatenating with repetition
print("\nMethod 9: With repetition")
result = str1 * 2 + " " + str2 * 2
print("Result:", result)

File: DSA/String/Code/ConcatenatingTwoStrings.cpp
#include <iostream>
#include <string>
using namespace std;

// Concatenating Two Strings in C++
// Join two strings together

int main() {
    string str1 = "Hello";
    string str2 = "World";
    
    // Method 1: Using + operator
    cout << "Method 1: Using + operator\n";
    string result1 = str1 + " " + str2;
    cout << "Result: " << result1 << "\n";
    
    // Method 2: Using += operator
    cout << "\nMethod 2: Using += operator\n";
    string result2 = str1;
    result2 += " ";
    result2 += str2;
    cout << "Result: " << result2 << "\n";
    
    // Method 3: Using append() method
    cout << "\nMethod 3: Using append()\n";
    string result3 = str1;
    result3.append(" ");
    result3.append(str2);
    cout << "Result: " << result3 << "\n";
    
    // Method 4: Using concatenate function
    cout << "\nMethod 4: Using string constructor\n";
    string result4 = string(str1) + " " + string(str2);
    cout << "Result: " << result4 << "\n";
    
    // Method 5: Concatenating multiple strings
    cout << "\nMethod 5: Multiple strings\n";
    string str3 = "from";
    string str4 = "C++";
    string result5 = str1 + " " + str2 + " " + str3 + " " + str4;
    cout << "Result: " << result5 << "\n";
    
    // Method 6: Using insert()
    cout << "\nMethod 6: Using insert()\n";
    string result6 = str1;
    result6.insert(str1.length(), " " + str2);
    cout << "Result: " << result6 << "\n";
    
    return 0;
}
